src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
```

Comment:
Also, I think we're getting a little long with our variable/parm/arg names.  Let's just call it an `index_name` rather than `search_index_name` and I think `search_index` would be better named as `search_func` to sort mirror the `View` functions.  After all the `search_index` really is a function, right?